.travis.yml

```diff
@@ -24,134 +36,117 @@ env:
         DEP_OPTS="NO_QT=1"
-        GOAL="install"
+        RUN_TESTS=false
+        GOAL=install
         BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports"
-# Win32
-    - >-
+    - stage: test
+      os: linux
+      env: >-
         HOST=i686-w64-mingw32
         DPKG_ADD_ARCH="i386"
         DEP_OPTS="NO_QT=1"
         PACKAGES="python3 nsis g++-mingw-w64-i686 wine-binfmt wine32"
-        RUN_TESTS=true
-        GOAL="install"
+        GOAL=install
         BITCOIN_CONFIG="--enable-reduce-exports"
-# Win64
-    - >-
+    - stage: test
+      os: linux
+      env: >-
         HOST=x86_64-w64-mingw32
         DEP_OPTS="NO_QT=1"
         PACKAGES="python3 nsis g++-mingw-w64-x86-64 wine-binfmt wine64"
-        RUN_TESTS=true
-        GOAL="install"
+        GOAL=install
         BITCOIN_CONFIG="--enable-reduce-exports"
-# 32-bit + dash
-    - >-
+    - stage: test
+      os: linux
+      env: >-
         HOST=i686-pc-linux-gnu
         PACKAGES="g++-multilib python3-zmq"
         DEP_OPTS="NO_QT=1"
-        RUN_TESTS=true
-        GOAL="install"
+        GOAL=install
         BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports LDFLAGS=-static-libstdc++"
-        CONFIG_SHELL="/bin/dash"
-# x86_64 Linux (uses qt5 dev package instead of depends Qt to speed up build and avoid timeout)
-    - >-
+        CONFIG_SHELL=/bin/dash
+    - stage: test
+      os: linux
+      env: >-
         HOST=x86_64-unknown-linux-gnu
         PACKAGES="python3-zmq qtbase5-dev qttools5-dev-tools protobuf-compiler libdbus-1-dev libharfbuzz-dev libprotobuf-dev"
         DEP_OPTS="NO_QT=1 NO_UPNP=1 DEBUG=1 ALLOW_HOST_PACKAGES=1"
-        RUN_TESTS=true
-        GOAL="install"
+        GOAL=install
         BITCOIN_CONFIG="--enable-zmq --with-gui=qt5 --enable-glibc-back-compat --enable-reduce-exports --enable-debug CXXFLAGS=\"-g0 -O2\""
-# x86_64 Linux (Qt5 & system libs)
-    - >-
+    - stage: test
+      os: linux
+      env: >-
         HOST=x86_64-unknown-linux-gnu
         PACKAGES="python3-zmq qtbase5-dev qttools5-dev-tools libssl1.0-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libprotobuf-dev protobuf-compiler libqrencode-dev"
         NO_DEPENDS=1
-        RUN_TESTS=true
-        GOAL="install"
+        GOAL=install
         BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --enable-glibc-back-compat --enable-reduce-exports --with-gui=qt5 CPPFLAGS=-DDEBUG_LOCKORDER"
-# x86_64 Linux, No wallet
-    - >-
+    - stage: test
+      os: linux
+      env: >-
         HOST=x86_64-unknown-linux-gnu
-        PACKAGES="python3"
-        DEP_OPTS="NO_WALLET=1"
-        RUN_TESTS=true
+        PACKAGES="python3" DEP_OPTS="NO_WALLET=1"
         GOAL="install"
         BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports"
-# Cross-Mac
-    - >-
-        HOST=x86_64-apple-darwin14
```

Comment:
I'd prefer if we kept the cross compile on linux, since something similar is used for producing the release binaries.

Reply:
I see. So it should cross-build on linux and for the purpose of this PR do the build on osx additionally, not instead of!

Reply:
Yes, both would be fine with me.